### PR TITLE
Implement integration tests

### PR DESCRIPTION
**Pull request by ihhyy**

### CODE REVIEW

File: IntegrationTests/Tests/BasicTests.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+using System;
+using System.Configuration;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace IntegrationTest.Tests
+{
+    public class BasicTests
```
Comment (denja2) [inline IntegrationTests/Tests/BasicTests.cs]: classnameIntegrationTests

File: IntegrationTests/Tests/BasicTests.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        [InlineData("oslo")]
+        [InlineData("LONDON")]
+        [InlineData("ToKyO")]
+        [InlineData("Toronto")]
+        public async void GetWeatherAsync_CorrectInput_ReturnMessageWithData(string input)
+        {
+            //Arrange
+
+            //Act
+            var output = await _weatherService.GetWeatherByCytyNameAsync(input);
```
Comment (denja2) [inline IntegrationTests/Tests/BasicTests.cs]: rename cyty

File: IntegrationTests/Tests/BasicTests.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            var match = Regex.Match(output.Message, @"(\w?)(.*?)\.| (\B\W)(.*?)\.", RegexOptions.IgnorePatternWhitespace).Success;
```
Comment (denja2) [inline IntegrationTests/Tests/BasicTests.cs]: to arrange, use correct regex 'In {CityName} /d now. {Comment|dasd|fsd}

File: IntegrationTests/Tests/BasicTests.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+            //Assert
+
+            Assert.NotNull(output.Message);
+            Assert.True(match);
+            Assert.True(!output.IsBadRequest);
```
Comment (denja2) [inline IntegrationTests/Tests/BasicTests.cs]: use false

File: IntegrationTests/Tests/BasicTests.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        }
+
+        [Fact]
+        public async Task GetWeatherAsync_EmptytInput_ReturnExceptionAsync()
```
Comment (denja2) [inline IntegrationTests/Tests/BasicTests.cs]: name

File: IntegrationTests/Tests/BasicTests.cs [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+            //Act
+            var output = await Assert.ThrowsAsync<EmptyInputException>(() => _weatherService.GetWeatherByCytyNameAsync(input));
+
+            //Assert
+            Assert.Equal(message, output.Message);
+        }
+
+        [Fact]
+        public async void GetWeatherAsync_EmptytInput_ReturnMessageWithoutDataAsync()
```
Comment (denja2) [inline IntegrationTests/Tests/BasicTests.cs]: name

File: IntegrationTests/Tests/BasicTests.cs (hunk 1 above)
Comment (ihhyy) [reply IntegrationTests/Tests/BasicTests.cs]: Fixed

File: IntegrationTests/Tests/BasicTests.cs (hunk 2 above)
Comment (ihhyy) [reply IntegrationTests/Tests/BasicTests.cs]: Renamed

File: IntegrationTests/Tests/BasicTests.cs (hunk 4 above)
Comment (ihhyy) [reply IntegrationTests/Tests/BasicTests.cs]: Done

File: IntegrationTests/Tests/BasicTests.cs (hunk 5 above)
Comment (ihhyy) [reply IntegrationTests/Tests/BasicTests.cs]: Fixed

File: IntegrationTests/Tests/BasicTests.cs (hunk 6 above)
Comment (ihhyy) [reply IntegrationTests/Tests/BasicTests.cs]: Fixed

File: IntegrationTests/Tests/BasicTests.cs (hunk 3 above)
Comment (ihhyy) [reply IntegrationTests/Tests/BasicTests.cs]: Remade

File: IntegrationTests/Tests/GetWeatherByCityNameIntegrationTests.cs [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        private readonly IValidator _validator;
+        private readonly IWeatherService _weatherService;
+        private readonly string _key;
+        private readonly string _API;
+
+        public GetWeatherByCityNameIntegrationTests()
+        {
+            _config = new TestConfig();
+            _key = _config.InitConfiguration()["APIkey"];
+            _API = _config.InitConfiguration()["urlCurrent"];
```
Comment (denja2) [inline IntegrationTests/Tests/GetWeatherByCityNameIntegrationTests.cs]: what is the reason to instantiate config twice

File: IntegrationTests/Tests/GetWeatherByCityNameIntegrationTests.cs [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        [InlineData("ToKyO")]
+        [InlineData("Toronto")]
+        public async void GetWeatherAsync_CorrectInput_ReturnMessageWithData(string input)
+        {
+            //Arrange
+            var appendix1 = " Dress warm";
+            var appendix2 = " It's fresh";
+            var appendix3 = " Good weather";
+            var appendix4 = " It's time to go to the beach";
+            var regerx = new Regex(@".?\d+.\d+");
```
Comment (denja2) [inline IntegrationTests/Tests/GetWeatherByCityNameIntegrationTests.cs]: check the name

File: IntegrationTests/Tests/GetWeatherByCityNameIntegrationTests.cs [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+            //Act
+            var output = await _weatherService.GetWeatherByCityNameAsync(input);
+            var message = $"In city {input} {regerx} °C now.{appendix1} || {appendix2} || {appendix3} || {appendix4}";
```
Comment (denja2) [inline IntegrationTests/Tests/GetWeatherByCityNameIntegrationTests.cs]: to arrange section

File: IntegrationTests/Tests/GetWeatherByCityNameIntegrationTests.cs (hunk 8 above)
Comment (ihhyy) [reply IntegrationTests/Tests/GetWeatherByCityNameIntegrationTests.cs]: Fixed

**Pull request merged**